<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Template Matching Params -->
    <arg name="match_frame"             default="base_footprint"/>
    <arg name="template_files_path"     default="$(find thorp_perception)/meshes"/>
    <arg name="output_frame"            default="template_match"/>
    <arg name="visualize_output"        default="false"/>
    <arg name="debug"                   default="false"/>
    <arg name="latch_initial_estimate"  default="false"/>
    <arg name="provide_processed_cloud" default="true"/>

    <!-- Launch template matching node -->
    <node pkg="rail_mesh_icp" type="template_matcher_node" name="template_matcher" output="screen" respawn="true">
      <param name="matching_frame"      value="$(arg match_frame)"/>
      <param name="template_files_path" value="$(arg template_files_path)"/>
      <param name="template_frame"      value="$(arg output_frame)"/>
      <param name="visualize"           value="$(arg visualize_output)"/>
      <param name="debug"               value="$(arg debug)"/>
      <param name="latch_initial"       value="$(arg latch_initial_estimate)"/>
      <param name="pre_processed_cloud" value="$(arg provide_processed_cloud)"/>

      <!-- ICP matcher parameters -->
      <param name="iterations"          value="50"/>
      <param name="max_distance"        value="0.01"/>
      <param name="trans_epsilon"       value="1e-8"/>
      <param name="fit_epsilon"         value="1e-8"/>
    </node>
</launch>
